aws/rust-runtime/aws-sigv4/src/http_request/canonical_request.rs

```diff
@@ -170,10 +179,19 @@ impl<'a> CanonicalRequest<'a> {
                     .expect("presigning requires expires_in")
                     .as_secs()
                     .to_string(),
-                security_token: params.security_token,
+                security_token,
                 signed_headers,
             }),
         };
+
+        if params.settings.session_token_mode == SessionTokenMode::Exclude {
+            if let Some(security_token) = params.security_token {
+                let mut sec_header = HeaderValue::from_str(security_token)?;
+                sec_header.set_sensitive(true);
+                canonical_headers.insert(header::X_AMZ_SECURITY_TOKEN, sec_header);
+            }
+        }
```

Comment:
can we handle this in `Self::headers` instead of mutating the headers later?

Reply:
I just realized this part is redundant. It's already in the `Self::headers` function. The header is added to the canonical headers in all cases.

I'll remove the mode condition.